code/modules/modular_computers/file_system/programs/system/file_browser.dm

```diff
@@ -28,28 +39,48 @@
 		var/obj/item/weapon/computer_hardware/hard_drive/HDD = computer.hard_drive
 		if(!HDD)
 			return 1
+		var/newpassword = sanitize(input(usr, "Enter a password or leave blank to leave unprotected:"))
 		var/datum/computer_file/data/F = new/datum/computer_file/data()
 		F.filename = newname
 		F.filetype = "TXT"
+		F.password = newpassword
 		HDD.store_file(F)
 	if(href_list["PRG_deletefile"])
 		. = 1
 		var/obj/item/weapon/computer_hardware/hard_drive/HDD = computer.hard_drive
 		if(!HDD)
 			return 1
 		var/datum/computer_file/file = HDD.find_file_by_name(href_list["PRG_deletefile"])
+		var/checkpass = ""
+		var/currentpass = file.password
 		if(!file || file.undeletable)
 			return 1
-		HDD.remove_file(file)
+		if(!currentpass)
```

Comment:
deleting shouldn't require password

Reply:
I disagree, bald greytide names steve breaks into the captains office and logs onto his computer, baldie mcsteve finds file names "realy important information i need to remember and cannot be deleted" baldie mcsteve deleted the file and runs to his greytide layer. The captain returns and finds his file has been deleted.

Reply:
Password protected files IRL ask for a password every time you so much as breathe on it. It requires a password when you move the file or delete the file. It makes sense that password protected files ask for a password every time you modify it.

Reply:
It would depend on what the actual underlying mentality of the password protection is. There's two ways to handle password protection:

- OS level. This is what you're referring to, in that the OS enforces password protection to even breathe on the file.
- Encrypted file. This just means that the contents of the file are encrypted and require a key to decrypt. You can move/copy/delete the file without issues, tho. Because it's just a random text file with garbled contents.

I don't really have a preference, @Arrow768 might have one.

Reply:
Regarding password required for deletion:
Most operating systems do not care weather a file is encrypted or not. The actual encryption / decryption is handled by a application interacting with the file.
The operating system only uses RBAC or something simmilar, but there are no special checks if a file is encrypted or not.

I believe the second option is our way to go in that case:
Viewing / editing requires a key (saved with the file datum)
but moving / copying / deleting does not.
This leaves room to add file owners and permissions later on.

Reply:
Regarding Baldie McSteve.
Physical Security is a important aspect.
If you allow someone else physical access to a computer system, then they can do almost anything with it.
Therefore I would allow other people to tamper with it, if they can get access.
It should be the responsibility of the player to restrict the access to the computer system accordingly or keep backups